From a4f3197b7ec2655fc27bb8c7168ff81f49245514 Mon Sep 17 00:00:00 2001
From: PratikBhusal <PratikBhusal@users.noreply.github.com>
Date: Mon, 24 May 2021 22:44:27 -0500
Subject: [PATCH] Invert fg/bg selection when they are the same

This patch is the exact same as the 20180309-c5ba9c0 patch by dequis -
dx (at) dxzone (dot) com (dot) ar

Because it follows the "git --format-patch" file format, you may also
use "git am" to apply this patch.
---
 x.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/x.c b/x.c
index 2a3bd38..c936d74 100644
--- a/x.c
+++ b/x.c
@@ -1452,9 +1452,14 @@ xdrawglyphfontspecs(const XftGlyphFontSpec *specs, Glyph base, int len, int x, i
 	}
 
 	if (base.mode & ATTR_REVERSE) {
-		temp = fg;
-		fg = bg;
-		bg = temp;
+		if (bg == fg) {
+			bg = &dc.col[defaultfg];
+			fg = &dc.col[defaultbg];
+		} else {
+			temp = fg;
+			fg = bg;
+			bg = temp;
+		}
 	}
 
 	if (base.mode & ATTR_BLINK && win.mode & MODE_BLINK)
-- 
2.43.0

